### PR TITLE
SDK Schedule Policy

### PR DESCRIPTION
**Pull request by ipochi**
**What this PR does / why we need it**:
This PR Adds SchedulePolicy api's to osd-sdk
**Which issue(s) this PR fixes** (optional)
Part of #407

**Special notes for your reviewer**:
Generated Files are in a separate commit.
SpecHandler code in type SchedulePolicyServer is commented out for clarification.

OSD Rest server has the below interface
https://github.com/libopenstorage/openstorage/blob/055b88ca745e1bf046f0cb8d8ba96899f3bf66e8/schedpolicy/sched_policy.go#L10

but the same Enumerate and Get calls for OSD-SDK  , it has been clubbed as one.

### CODE REVIEW

File: api/server/sdk/schedule_policy.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		return nil, status.Errorf(
+			codes.Internal,
+			"failed to create schedule policy: %v",
+			err.Error())
+	}
+
+	return &api.SdkSchedulePolicyCreateResponse{}, err
+}
+
+// Update method creates schedule policy
```
Comment (ram-infrac) [inline api/server/sdk/schedule_policy.go]: nit: updates schedule policy.
Similar changes for rest of comments

File: api/api.proto [hunk 2]
```diff
@@ -911,6 +911,86 @@ service OpenStorageCredentials {
...
+}
+
+message SdkSchedulePolicyDeleteResponse {
+}
+
+message SdkSchedulePolicy {
+  // Name of the schedule policy
+  string name = 1;
+  // Schedule string in yaml
+  string schedule = 2;
```
Comment (lpabon) [inline api/api.proto]: This should probably not be a string. Looks like multiple values are provided encoded here. Not part of this PR, but something that needs to be addressed before release.

File: api/server/sdk/schedule_policy.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+	"github.com/libopenstorage/openstorage/api"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// SchedulePolicyServer is an implementation of the gRPC OpenStorageSchedulePolicy interface
+type SchedulePolicyServer struct {
+	api.OpenStorageSchedulePolicyServer
+	//specHandler spec.SpecHandler
```
Comment (lpabon) [inline api/server/sdk/schedule_policy.go]: If not needed, please remove.

File: api/server/sdk/schedule_policy.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	cluster cluster.Cluster
+}
+
+// Create method creates schedule policy
+func (s *SchedulePolicyServer) Create(
+	ctx context.Context,
+	req *api.SdkSchedulePolicyCreateRequest,
+) (*api.SdkSchedulePolicyCreateResponse, error) {
+
+	if len(req.GetSchedulePolicy().GetName()) == 0 {
```
Comment (lpabon) [inline api/server/sdk/schedule_policy.go]: Need check first if `req.GetSchedulePolicy() != nil`

Comment (lpabon) [reply api/server/sdk/schedule_policy.go]: You will need to do this in all the functions that use this object.

File: api/server/sdk/schedule_policy.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+			codes.Internal,
+			"failed to delete schedule policy: %v",
+			err.Error())
+	}
+
+	return &api.SdkSchedulePolicyDeleteResponse{}, err
+}
+
+// Enumerate method enumerates schedule policy
+func (s *SchedulePolicyServer) Enumerate(
```
Comment (lpabon) [inline api/server/sdk/schedule_policy.go]: This API does not make sense to me. The Golang interface 

https://github.com/libopenstorage/openstorage/blob/055b88ca745e1bf046f0cb8d8ba96899f3bf66e8/schedpolicy/sched_policy.go#L17-L18

How can it get `the ones specified` if it takes no arguments?

I think that is incorrect. Enumerate should just return them all since there are no "flags" to match. No need to take `Name` as an argument.

What do you guys think @ipochi @ram-infrac ?

File: api/api.proto (hunk 2 above)
Comment (ipochi) [reply api/api.proto]: from what  i recall , this is a yaml , which is passed as a string.

Comment (lpabon) [reply api/api.proto]: Created https://github.com/libopenstorage/openstorage/issues/459 to track
